internal/cluster/cluster.go

```diff
@@ -777,6 +778,24 @@ func (m *Manager) DeleteClusterFiles(ctx context.Context, c *common.Cluster, obj
 	return nil
 }
 
+func (m Manager) InactiveClusterDeregister(ctx context.Context, inactiveSince strfmt.DateTime, objectHandler s3wrapper.API) error {
+	var clusters []*common.Cluster
+	db := m.db.Unscoped()
+	if reply := db.Where("updated_at < ?", inactiveSince).Find(&clusters); reply.Error != nil {
+		return reply.Error
+	}
+	for i := range clusters {
+		c := clusters[i]
+		m.eventsHandler.AddEvent(ctx, *c.ID, nil, models.EventSeverityError,
+			"Cluster is deregistered due to inactivity", time.Now())
```

Comment:
Maybe add current state?

Reply:
cluster id?

Reply:
same as the previous comment, the event already has this information, I don't see the point of having the id copied and get a less searchable event.